master/internal/kubernetes/request_queue_test.go

```diff
@@ -0,0 +1,211 @@
+package kubernetes
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	petName "github.com/dustinkirkland/golang-petname"
+	"gotest.tools/assert"
+
+	"github.com/determined-ai/determined/master/pkg/actor"
+
+	k8sV1 "k8s.io/api/core/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	typedV1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+type (
+	mockPodActorPing struct{}
+	deleteMockPod    struct{}
+)
+
+type mockPodActor struct {
+	requestQueue *actor.Ref
+	name         string
+}
+
+func newMockPodActor(requestQueue *actor.Ref) *mockPodActor {
+	return &mockPodActor{
+		requestQueue: requestQueue,
+		name:         petName.Generate(3, "-"),
+	}
+}
+
+func (m *mockPodActor) Receive(ctx *actor.Context) error {
+	switch msg := ctx.Message().(type) {
+	case actor.PreStart:
+		podSpec := k8sV1.Pod{ObjectMeta: metaV1.ObjectMeta{Name: m.name}}
+		cmSpec := k8sV1.ConfigMap{ObjectMeta: metaV1.ObjectMeta{Name: m.name}}
+
+		ctx.Tell(m.requestQueue, createKubernetesResources{
+			handler:       ctx.Self(),
+			podSpec:       &podSpec,
+			configMapSpec: &cmSpec,
+		})
+
+	case mockPodActorPing:
+		ctx.Respond(mockPodActorPing{})
+
+	case deleteMockPod:
+		ctx.Ask(m.requestQueue, deleteKubernetesResources{
+			handler:       ctx.Self(),
+			podName:       m.name,
+			configMapName: m.name,
+		})
+
+	case resourceCreationCancelled:
+
+	case resourceCreationFailed, resourceDeletionFailed:
+		ctx.Log().Errorf("should not hit these messages during testing %T", msg)
+		return actor.ErrUnexpectedMessage(ctx)
+
+	default:
+		ctx.Log().Errorf("unexpected message %T", msg)
+		return actor.ErrUnexpectedMessage(ctx)
+	}
+
+	return nil
+}
+
+func getNumberOfActivePods(podInterface typedV1.PodInterface) int {
+	podList, err := podInterface.List(metaV1.ListOptions{})
+	if err != nil {
+		panic(err)
+	}
+
+	return len(podList.Items)
+}
+
+func waitForPendingRequestToFinish(k8RequestQueue *requestQueue) {
+	time.Sleep(time.Second)
+
+	// Wait for queue to finish all in flight requests.
+	for len(k8RequestQueue.queue) > 0 &&
```

Comment:
blocking: This has a race condition. Probably write a loop to wait for it to finish, which might be more robust because you don't know how much time it takes on CI machines.

Reply:
not exactly sure what you have in mind for a loop, as this is already a loop. I do agree this isn't great overall but really don't see a better way to test without adding test-only messages to the actors.

Reply:
ah my bad. for some reason, I thought it was an if statement. :( I mean adding a sleep statement inside the for a loop.

Reply:
added a sleep in